dev/fattest.simplicity/src/componenttest/custom/junit/runner/TestNameFilter.java

```diff
@@ -55,13 +92,47 @@ public String describe() {
 
     @Override
     public boolean shouldRun(Description arg0) {
-        if (FAT_TEST_CLASS != null && !arg0.getClassName().equals(FAT_TEST_CLASS)) {
+        if (FAT_TEST_CLASS != null && !match(FAT_TEST_CLASS, arg0.getClassName())) {
             return false;
         }
-        if (FAT_TEST_METHOD != null && !arg0.getMethodName().equals(FAT_TEST_METHOD)) {
+        if (FAT_TEST_METHOD != null && !match(FAT_TEST_METHOD, arg0.getMethodName())) {
             return false;
         }
         return true;
     }
 
+    private boolean match(String list, String arg) {
+        boolean match = false;
+        for (String s : list.split(",")) {
+            if (s.contains(".")) {
+                if (compare(s, arg)) {
+                    match = true;
+                    break;
+                }
+            } else {
+                if (arg.contains(".")) {
+                    int indx = arg.lastIndexOf(".");
+                    arg = arg.substring(indx + 1);
+                }
+                if (compare(s, arg)) {
+                    match = true;
+                    break;
+                }
+            }
+        }
+        System.out.println(arg + " " + match);
+        return match;
+    }
+
+    private boolean compare(String pattern, String arg) {
+        if (pattern.contains("*")) {
+            return wildcardMatch(pattern, arg);
+        } else {
+            return arg.equals(pattern);
+        }
```

Comment:
instead of branching here we could just call `wildCardMatch` in all cases, and when there are no `*` it would be the same as .equals()